conda/environments/all.yaml

```diff
@@ -0,0 +1,34 @@
+# This file is generated by `rapids-dependency-file-generator`.
```

Comment:
All environment files need to be generated by `dependencies.yaml`.

The failure in pre-commit is because `rapids-dependency-file-generator` [looks for files that it generated and removes them](https://github.com/rapidsai/dependency-file-generator/blob/78ca9ecbf01edaff93f898f2202fb34342f3820e/src/rapids_dependency_file_generator/rapids_dependency_file_generator.py#L39-L40) before re-generating the new files. You have the header here saying it's a generated file, but there's no declaration of this environment file in `dependencies.yaml`.

This is how you tell it to generate a given environment file: https://github.com/rapidsai/ucxx/blob/4464ca942d5ac71aabe4492e2a6d59da453f06dc/dependencies.yaml#L3-L17

You might want to add a section like this for the C++ dependency list.
```
files:
  cpp:     # This is the filename
    output: conda
    matrix:
      arch: [x86_64]
    includes:
      - build_cpp
      - checks
      - dev
      - test_cpp
```

Reply:
Thanks Bradley, this is helpful. I'd still like to trim that more though, for example the above will result in:

```
# This file is generated by `rapids-dependency-file-generator`.
# To make changes, edit ../../dependencies.yaml and run `rapids-dependency-file-generator`.
channels:
- rapidsai
- rapidsai-nightly
- dask/label/dev
- conda-forge
- nvidia
dependencies:
- autoconf
- automake
- c-compiler
- cmake>=3.26.4
- cxx-compiler
- dask-cuda==23.12.*
- dask-cudf==23.12.*
- fmt>=9.1.0,<10
- gmock>=1.13.0
- gtest>=1.13.0
- librmm==23.12.*
- libtool
- ninja
- pip
- pkg-config
- pre-commit
- spdlog>=1.11.0,<1.12
name: cpp_arch-x86_64
```

That include a bunch of things we don't need: `rapidsai`/`rapidsai-nightly`/`dask/label/dev`/`nvidia` channels and `dask-cuda`/`dask-cudf`/`librmm`/`pip`/`pkg-config`. Is there a way to minimize that?

Additionally, this is supposed to be valid for `aarch64` too, can we make it architecture agnostic?

Reply:
There's currently no way to drop extra channels from the generated files. You can file an issue to rapids-dependency-file-generator if you have a proposal for how the `dependencies.yaml` schema should accommodate that idea, but I'm not sure if it's worth changing. https://github.com/rapidsai/dependency-file-generator

If you depend only on architecture-agnostic packages, you can just drop the arch from the `files:` spec.
```
files:
  cpp:     # This is used as part of the environment's filename
    output: conda
    includes:
      - build_cpp
      - checks
      - dev
      - test_cpp
```